### PR TITLE
Students project

### PR DESCRIPTION
**Pull request by lubnaabd**
relates to #67

### CODE REVIEW

**Review by ali-7 — APPROVED** [review]

File: public/js/studentProject.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
+
+const addStd = document.getElementById('addStd');
+const addStudent = document.getElementById('addStudent');
+const selector = document.getElementById('selector');
+const URL = window.location.href;
```
Comment (NouraldinS) [inline public/js/studentProject.js]: Rename to `url`.

File: public/js/studentProject.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+
+let projectTypeValue = '';
+if (projectType.toLowerCase() === 'clients') {
+  projectTypeValue = 0;
+} else if (projectType.toLowerCase() === 'community') {
+  projectTypeValue = 1;
+}
+
+const addStudents = () => {
```
Comment (NouraldinS) [inline public/js/studentProject.js]: Rename to `toggleAddStudents`.

File: public/js/studentProject.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  addStd.classList.toggle('sectionAddstd--visible');
+};
+
+deleteStudentButton.forEach((button) => {
+  const deleteProjectData = { std_project: button.id };
```
Comment (NouraldinS) [inline public/js/studentProject.js]: Rename `std_project` to `studentProject`.

File: public/js/studentProject.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
+
+const addStd = document.getElementById('addStd');
+const addStudent = document.getElementById('addStudent');
+const selector = document.getElementById('selector');
+const URL = window.location.href;
+const splitUrl = URL.split('/');
+const cohortId = splitUrl[splitUrl.length - 1];
+const projectId = splitUrl[splitUrl.length - 4];
+const deleteStudentButton = document.querySelectorAll('.delete');
```
Comment (NouraldinS) [inline public/js/studentProject.js]: Rename to `deleteStudentButtons`.

File: public/js/studentProject.js (hunk 3 above)
Comment (NouraldinS) [reply public/js/studentProject.js]: And anyway, how does it make sense to assign `studentProject` the value of `button.id`? What are you doing here?

Comment (NouraldinS) [reply public/js/studentProject.js]: @lubnaabd I dug up what you were trying to achieve, firstly, line 9 `const deleteStudentButton = document.querySelectorAll('.delete');` isn't selecting any buttons, it is selecting anchor elements. So it is either you change your anchors to buttons, or change the variable's name to `deleteStudentAnchors` and act accordingly, then you're saying:
```js
const deleteProjectData = { std_project: button.id };
```
What is `std_project`? in your `src/views/cohortStudents.hbs` on line 28 you've said:
```hbs
<a class="delete" id="{{std_id}}">Delete</a>
```
Which by the way needs to be renamed to `studentId`.
You're passing here a student's ID, then assigning it to `std_project`? Why? What made sense when you've made this?

File: public/js/studentProject.js [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+deleteStudentButton.forEach((button) => {
+  const deleteProjectData = { std_project: button.id };
+  const route = `/admin/${projectType}/${projectId}/projects/student/${cohortId}`;
+  const routeToRedirect = `/admin/${projectType}/${projectId}/projects/student/${cohortId}`;
+  deleteButtonFunction(button, route, routeToRedirect, deleteProjectData);
+});
+
+
+addStudent.addEventListener('click', () => {
+  const stdId = selector[selector.selectedIndex].value;
```
Comment (NouraldinS) [inline public/js/studentProject.js]: Rename to `studentId`.

File: src/controllers/studentsProject.js [hunk 6]
```diff
@@ -0,0 +1,80 @@
+const getStudentProjectQuery = require('../database/queries/getStudentProject');
+const getCohortStudentsQuery = require('../database/queries/getCohortStudents');
+const { deleteStdProjectQuery } = require('../database/queries/deleteStdProject');
```
Comment (NouraldinS) [inline src/controllers/studentsProject.js]: Rename to `deleteStudentProjectQuery`.

File: src/controllers/studentsProject.js [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+const { addStdProjectQuery } = require('../database/queries/addStdProject');
```
Comment (NouraldinS) [inline src/controllers/studentsProject.js]: Rename to `addStudentProjectQuery`.

File: src/controllers/studentsProject.js [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+    .then((results) => {
+      const result = { err: null, message: 'Student Deleted !' };
+      response.send(JSON.stringify(result));
+    })
+    .catch((err) => {
+      next(err);
+    });
+};
+
+exports.addStdProject = (req, response, next) => {
```
Comment (NouraldinS) [inline src/controllers/studentsProject.js]: Rename to `addStudentProject`.

File: src/controllers/studentsProject.js [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+      layout: 'error',
+      statusCode: 404,
+      errorMessage: 'Page not found',
+      style: ['error'],
+    });
+  }
+};
+
+
+exports.deleteStdProject = (req, response, next) => {
```
Comment (NouraldinS) [inline src/controllers/studentsProject.js]: Rename to `deleteStudentProject`.

File: src/database/queries/addStdProject.js [hunk 10]
```diff
@@ -0,0 +1,18 @@
+const dbconnection = require('../db_connection');
```
Comment (NouraldinS) [inline src/database/queries/addStdProject.js]: Rename to `dbConnection` and rename file to `addStudentProject`.

File: src/database/queries/deleteStdProject.js [hunk 11]
```diff
@@ -0,0 +1,17 @@
+const dbconnection = require('../db_connection');
```
Comment (NouraldinS) [inline src/database/queries/deleteStdProject.js]: Rename to `dbConnection` and rename file to `deleteStudentProject`.

File: src/views/cohortStudents.hbs [hunk 12]
```diff
@@ -22,10 +22,10 @@
         <td><a href="{{githublink}}">Click</a></td>
         <td><a href="{{img_url}}">Click</a></td>
         <td>
-          <a class="edit" href = "/admin/cohorts/{{cohort_id}}/editStudent/{{id}}"id="{{id}}">Edit</a>
+          <a class="edit" href = "/admin/cohorts/{{cohort_id}}/editStudent/{{std_id}}"id="{{std_id}}">Edit</a>
```
Comment (NouraldinS) [inline src/views/cohortStudents.hbs]: Rename `std_id` to `studentId`.

File: src/views/projectsPage.hbs [hunk 13]
```diff
@@ -35,13 +35,13 @@
             Delete </a>
         </td>
         <td onclick="students()">
-          <a class="student" href="javascript:;">
+          <a class="student" href="/admin/{{../projectsType}}/{{id}}/projects/student/{{cohort_id}}">
```
Comment (NouraldinS) [inline src/views/projectsPage.hbs]: Rename `cohort_id` to `cohortId`.

File: src/views/studentsProject.hbs [hunk 14]
```diff
@@ -0,0 +1,38 @@
...
+    <button class="button button2 buttonstd" onclick="addStudents()">ADD Student</button>
+    {{#if res.projectStudentsResult.length}}
+    <table class="tabel ">
+      <tr>
+        <th>Name</th>
+        <th>Delete</th>
+      </tr>
+      {{#each res.projectStudentsResult}}
+      <tr>
+        <td>{{student_name}}</td>
```
Comment (NouraldinS) [inline src/views/studentsProject.hbs]: Rename to `studentName`.

**Review by NouraldinS — CHANGES_REQUESTED** [review]
Do not abbreviate the names of your functions nor your variables.

**Review by HemaSAli — APPROVED** [review]

**Review by NouraldinS — APPROVED** [review]

**Pull request merged**